sway/input/cursor.c

```diff
@@ -523,19 +519,15 @@ static void handle_touch_motion(struct wl_listener *listener, void *data) {
 	cursor_handle_activity_from_device(cursor, event->device);
 
 	struct sway_seat *seat = cursor->seat;
-	struct wlr_seat *wlr_seat = seat->wlr_seat;
-	struct wlr_surface *surface = NULL;
 
 	double lx, ly;
 	wlr_cursor_absolute_to_layout_coords(cursor->cursor, event->device,
 			event->x, event->y, &lx, &ly);
-	double sx, sy;
-	node_at_coords(cursor->seat, lx, ly, &surface, &sx, &sy);
 
-	if (seat->touch_id == event->touch_id) {
-		seat->touch_x = lx;
-		seat->touch_y = ly;
+	seat->touch_x = lx;
+	seat->touch_y = ly;
```

Comment:
Why is this outside of the `seat->touch_id == event->touch_id` condition? It sounds like this potentially mixes up some touch points.

Reply:
We use the seat->touch_x/y as storage to the default seatop seatop_touch_motion implementation. As the seat->touch_id is the last touch down touch id, moving this inside the condition make both touches to use the last touch point positions.

But that feels more like a comfort choice, so I decided to revert it. So now seat->touch_x/y will only represent the last touch positions (as before). In the seatop_down, I then recompute lx and ly using the event x and y.

I let this change in a dedicated commit to make it easier for review. I'll fixup if everything looks okay to you.